File: BurgerBuilderBackend/BurgerBuilder.DTOs/Classes/IncomingOrder.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace BurgerBuilder.DTOs.Classes
{
    public class IncomingOrder
    {
        public Ingridients Ingridients { get; set; }
        public decimal Price { get; set; }
        public OrderData OrderData { get; set; }
        public string UserId { get; set; }
    }
}

File: burger-builder/src/Types/IncomingOrder.ts
import { BurgerInnerIngridientName } from "./BurgerInnerIngridientName";
import { BurgerInnerIngridientsDictionary } from "./BurgerInnerIngridientsDictionary";
import IOrderData from "./OrderData";

export default interface IIncomingOrder {
    ingridients: BurgerInnerIngridientsDictionary,
    price: number,
    orderData: IOrderData,
    userId: string
}